test/Atmos/EDMF/edmf_kernels.jl

```diff
@@ -898,6 +906,48 @@ function precompute(::EDMF, bl, args, ts, ::Flux{SecondOrder})
     K_h = K_m / Pr_t
     ρaw_up = vuntuple(i -> up[i].ρaw, N_up)
 
+    ρu_gm_tup = Tuple(gm.ρu)
+    ρa_en = gm.ρ * env.a
+    # TODO: Consider turbulent contribution:
+
+    e_kin_up = vuntuple(N_up) do i
+        FT(1 // 2) * (
+            (gm.ρu[1] * ρ_inv)^2 +
+            (gm.ρu[2] * ρ_inv)^2 +
+            (fix_void_up(up[i].ρa, up[i].ρaw / up[i].ρa))^2
+        )
+    end
+    e_kin_en =
+        FT(1 // 2) * ((gm.ρu[1] * ρ_inv)^2 + (gm.ρu[2] * ρ_inv)^2 + env.w)^2
+
+    e_tot_up = ntuple(i -> total_energy(e_kin_up[i], _grav * z, ts_up[i]), N_up)
+    h_tot_up = ntuple(i -> total_specific_enthalpy(ts_up[i], e_tot_up[i]), N_up)
+
+    e_tot_en = total_energy(e_kin_en, _grav * z, ts_en)
+    h_tot_en = total_specific_enthalpy(ts_en, e_tot_en)
+    h_tot_gm = total_specific_enthalpy(ts, gm.energy.ρe * ρ_inv)
+
+
+    massflux_h_tot = sum(
+        ntuple(N_up) do i
+            fix_void_up(
+                ρa_up[i],
+                ρa_up[i] *
+                (h_tot_gm - h_tot_up[i]) *
+                (gm.ρu[3] * ρ_inv - ρaw_up[i] / ρa_up[i]),
+            )
+        end,
+    )
+    massflux_h_tot +=
+        (ρa_en * (h_tot_gm - h_tot_en) * (ρu_gm_tup[3] * ρ_inv - env.w))
+
+    ρh_sgs_flux = SVector{3, FT}(
+        0,
+        0,
+        -env.a * K_h * 3 * en_dif.∇h_tot[3] + massflux_h_tot,
```

Comment:
Why is there a 3 here? Also, I think ρ is missing.

Reply:
you are right about density.
the '3' is here for the vertical component of the gradient as for now we only have vertical components of the SGS flux included